domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/frequencies/FrequencyTest.java

```diff
@@ -0,0 +1,227 @@
+
+        assertEquals("frequencies.txt", notice.getFilename());
+        assertEquals("headway_secs", notice.getNoticeSpecific(KEY_FIELD_NAME));
+        assertEquals(STRING_TEST, notice.getEntityId());
+        assertEquals(1, noticeCollection.size());
+    }
+
+    @Test
+    void createFrequencyWithInvalidHeadwaySecsShouldGenerateNotice() {
+        final Frequency.FrequencyBuilder underTest = new Frequency.FrequencyBuilder();
+
+        final EntityBuildResult<?> entityBuildResult = underTest.tripId(STRING_TEST)
+                .startTime(VALID_START_TIME)
+                .endTime(VALID_END_TIME)
+                .headwaySecs(-1800)
+                .exactTimes(VALID_EXACT_TIMES_INTEGER)
+                .build();
+
+        assertTrue(entityBuildResult.getData() instanceof List);
+        // This is designed so that method getData returns a list of notices. Thereby there is no need for cast check.
+        //noinspection unchecked
+        final List<IntegerFieldValueOutOfRangeNotice> noticeCollection =
+                (List<IntegerFieldValueOutOfRangeNotice>) entityBuildResult.getData();
+        final IntegerFieldValueOutOfRangeNotice notice = noticeCollection.get(0);
+
+        assertEquals("frequencies.txt", notice.getFilename());
+        assertEquals("headway_secs", notice.getNoticeSpecific(KEY_FIELD_NAME));
+        assertEquals(STRING_TEST, notice.getEntityId());
+        assertEquals(0, notice.getNoticeSpecific(KEY_RANGE_MIN));
+        assertEquals(Integer.MAX_VALUE, notice.getNoticeSpecific(KEY_RANGE_MAX));
+        assertEquals(-1800, notice.getNoticeSpecific(KEY_ACTUAL_VALUE));
+        assertEquals(1, noticeCollection.size());
+    }
+
+    @Test
+    void createFrequencyWithInvalidExactTimesShouldGenerateNotice() {
+        final Frequency.FrequencyBuilder underTest = new Frequency.FrequencyBuilder();
+
+        final EntityBuildResult<?> entityBuildResult = underTest.tripId(STRING_TEST)
+                .startTime(VALID_START_TIME)
+                .endTime(VALID_END_TIME)
+                .headwaySecs(VALID_HEADWAY_SECS)
+                .exactTimes(2)
+                .build();
+
+        assertTrue(entityBuildResult.getData() instanceof List);
+        // This is designed so that method getData returns a list of notices. Thereby there is no need for cast check.
+        //noinspection unchecked
+        final List<UnexpectedEnumValueNotice> noticeCollection =
+                (List<UnexpectedEnumValueNotice>) entityBuildResult.getData();
+        final UnexpectedEnumValueNotice notice = noticeCollection.get(0);
+
+        assertEquals("frequencies.txt", notice.getFilename());
+        assertEquals("exact_times", notice.getNoticeSpecific(KEY_FIELD_NAME));
+        assertEquals(STRING_TEST, notice.getEntityId());
+        assertEquals(2, notice.getNoticeSpecific(KEY_ENUM_VALUE));
+        assertEquals(1, noticeCollection.size());
+    }
+
+    @Test
+    void createFrequencyWithNullTransfersShouldNotGenerateNotice() {
```

Comment:
method should be renamed `createFrequencyWithNullExactTimeShouldNotGenerateNotice`